### PR TITLE
on the user's profile, make the tip button open the QR code for the BCH address and remove the PayPal form. #1820

### PR DESCRIPTION
**Pull request by Sarthak5598**
on the user's profile, make the tip button open the QR code for the BCH address and remove the PayPal form. #1820

### CODE REVIEW

File: website/templates/includes/sidenav.html [hunk 1]
```diff
@@ -67,6 +67,14 @@
 		</ul>
 
 		
+		<ul class="space-y-3 font-medium text-3xl mt-8 mb-0">
+			<li class="mb-[-6px] ms-[1px]">
+				<span class="font-bold text-[#AAA] text-[1.2rem] tracking-wide">LAB</span>
```
Comment (DonnieBLT) [inline website/templates/includes/sidenav.html:72]: Can take this off and say "feedback" below

File: website/templates/profile.html [hunk 2]
```diff
@@ -209,14 +218,16 @@ <h1 class="page-header text-capitalize">{{ user.username }}</h1>
             {% endif %}
 
             <div class="status">
-                <form name="_xclick" action="https://www.paypal.com/cgi-bin/webscr" method="post">
+                <form id="qr-form">
+                    {% csrf_token %}
                     <input type="hidden" name="cmd" value="_xclick">
-                    <input type="hidden" name="business" value="{{ project.paypal|default:" coderbounty@gmail.com" }}">
+                    <input type="hidden" name="business" value="{{ project.BCH|default:"coderbounty@gmail.com" }}">
```
Comment (DonnieBLT) [inline website/templates/profile.html:224]: You can take all this previous form info out.  This doesn't need to be a form.

File: website/views.py [hunk 3]
```diff
@@ -3849,3 +3860,31 @@ def get_context_data(self, **kwargs):
...
+        img = qr.make_image(fill_color="black", back_color="white")
+        buffer = BytesIO()
+        img.save(buffer, format='PNG')
+        qr_image = buffer.getvalue()
+
+        # Encode the image data as base64
+        qr_image_base64 = base64.b64encode(qr_image).decode('utf-8')
+
+        # Return the base64-encoded image data as a JSON response
+        return JsonResponse({'qr_image': qr_image_base64})
```
Comment (DonnieBLT) [inline website/views.py:3888]: If you make this return an image type you can eliminate the JavaScript code and just use <img src=...

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Some feedback

File: website/forms.py [hunk 4]
```diff
@@ -27,4 +27,7 @@ class CaptchaForm(forms.Form):
 class QuickIssueForm(forms.Form):
     url = forms.CharField()
     label = forms.CharField()
-    description = forms.CharField()
\ No newline at end of file
+    description = forms.CharField()
+
+class BitcoinAddressForm(forms.Form):
```
Comment (DonnieBLT) [inline website/forms.py:32]: Don't need this

File: website/templates/includes/sidenav.html [hunk 5]
```diff
@@ -67,6 +67,14 @@
 		</ul>
 
 		
+		<ul class="space-y-3 font-medium text-3xl mt-8 mb-0">
+			<li class="mb-[-6px] ms-[1px]">
+				<span class="font-bold text-[#AAA] text-[1.2rem] tracking-wide">Feedback</span>
+			</li>
```
Comment (DonnieBLT) [inline website/templates/includes/sidenav.html:73]: Put this in another pr

File: website/templates/profile.html [hunk 6]
```diff
@@ -209,15 +218,11 @@ <h1 class="page-header text-capitalize">{{ user.username }}</h1>
...
-                    <input type="hidden" name="item_name" value="tip for {{ user.username }} on {% env 'DOMAIN_NAME' %}">
-                    <input type="hidden" name="currency_code" value="USD">
-                    <button type="submit" class="btn btn-danger" style="color: black;">
-                        <i class="fa fa-money fa-lg"></i> {% trans 'Send a tip' %}
-                    </button>
-                </form>
+                <button type="button" id="openModal" class="btn btn-danger" style="color: black;">Send a Tip</button>
+                <dialog id="dialog">
+                    <img id="qr-image" src="{% url 'generate_bch_qr' %}" alt="QR Code">
+                    <p id="bch-address">{{ request.user.userprofile.crypto_address }}</p>
```
Comment (DonnieBLT) [inline website/templates/profile.html:224]:
```suggestion
                    <p id="bch-address">BCH: {{ request.user.userprofile.crypto_address }}</p>
```

File: website/views.py [hunk 7]
```diff
@@ -91,6 +95,13 @@
 from django.http import HttpRequest
 from django.utils.timezone import now
 
+
+from django.shortcuts import render
+from .forms import BitcoinAddressForm
```
Comment (DonnieBLT) [inline website/views.py:100]: Not used

File: website/views.py [hunk 8]
```diff
@@ -3849,3 +3860,30 @@ def get_context_data(self, **kwargs):
 #                     headers=headers,
 #                 )
 #         mail.logout()
+
+from django.http import HttpResponse
```
Comment (DonnieBLT) [inline website/views.py:3864]: Duplicate imports?

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Some tweaks

File: website/templates/includes/sidenav.html (hunk 5 above)
Comment (Sarthak5598) [reply website/templates/includes/sidenav.html:73]: I have already made a pr on this please check I'll remove it from here

**Review by DonnieBLT — APPROVED** [review]

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Can you please resolve the conflicts?

**Comment by Sarthak5598** [comment]
@DonnieBLT  please check it out

File: blt/urls.py [hunk 9]
```diff
@@ -412,15 +412,10 @@
     re_path(r"^ratings/", include("star_ratings.urls", namespace="ratings")),
     path("robots.txt", website.views.robots_txt),
     path("ads.txt", website.views.ads_txt),
-    re_path(r"^contributors/$", contributors_view, name="contributors"),
-    path("company/", include("company.urls")),
-    path("sponsor/", website.views.sponsor_view, name="sponsor"),
-    path("companies/", DomainListView.as_view(), name="domain_list"),
```
Comment (DonnieBLT) [inline blt/urls.py:418]: What happened here?

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
See comment

File: blt/urls.py (hunk 9 above)
Comment (Sarthak5598) [reply blt/urls.py:418]: There were Conflicts here
should I add them back ?

Comment (DonnieBLT) [reply blt/urls.py:418]: Yes please

Comment (Sarthak5598) [reply blt/urls.py:418]: Done

File: website/views.py [hunk 10]
```diff
@@ -12,15 +12,12 @@
 from datetime import datetime, timedelta, timezone
 from decimal import Decimal
 from urllib.parse import urlparse, urlsplit, urlunparse
+from io import BytesIO
 
-import humanize
 import requests
 import requests.exceptions
 import six
-import stripe
```
Comment (DonnieBLT) [inline website/views.py:20]: Please check your full PR

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
See comment

File: website/views.py (hunk 10 above)
Comment (Sarthak5598) [reply website/views.py]: I have made all the changes

File: website/views.py [hunk 11]
```diff
@@ -92,6 +93,10 @@
 from .forms import CaptchaForm, HuntForm, QuickIssueForm, UserProfileForm
 
 
+
+import qrcode
+from django.http import HttpResponse
```
Comment (DonnieBLT) [inline website/views.py:98]: This is already imported on line 51

File: website/views.py [hunk 12]
```diff
@@ -92,6 +93,10 @@
 from .forms import CaptchaForm, HuntForm, QuickIssueForm, UserProfileForm
 
 
+
+import qrcode
```
Comment (DonnieBLT) [inline website/views.py:97]: Please move this up to the other imports

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Two tweaks

**Comment by Sarthak5598** [comment]
resolved

File: website/templates/profile.html [hunk 13]
```diff
@@ -209,15 +210,11 @@ <h1 class="page-header text-capitalize">{{ user.username }}</h1>
...
-                    <input type="hidden" name="item_name" value="tip for {{ user.username }} on {% env 'DOMAIN_NAME' %}">
-                    <input type="hidden" name="currency_code" value="USD">
-                    <button type="submit" class="btn btn-danger" style="color: black;">
-                        <i class="fa fa-money fa-lg"></i> {% trans 'Send a tip' %}
-                    </button>
-                </form>
+                <button type="button" id="openModal" class="btn btn-danger" style="color: black;">Send a Tip</button>
+                <dialog id="dialog">
+                    <img id="qr-image" src="{% url 'generate_bch_qr' username=request.user.username %}" alt="QR Code">
+                    <p id="bch-address" style="color: aliceblue;">BCH: {{ request.user.userprofile.crypto_address }}</p>
```
Comment (DonnieBLT) [inline website/templates/profile.html:216]: This is still current users

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Update

File: website/templates/profile.html (hunk 13 above)
Comment (Sarthak5598) [reply website/templates/profile.html:216]: can you elaborate on what you're looking for I am really confused

Comment (DonnieBLT) [reply website/templates/profile.html:216]: This should be the address of the user, who we are requesting the address of not the current request.user

Comment (DonnieBLT) [reply website/templates/profile.html:216]: Looks like it

Comment (Sarthak5598) [reply website/templates/profile.html]: Commited the change please check it out

**Comment by DonnieBLT** [comment]
looks like a conflict, can you please fix it

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
This looks good, other than the conflicts. Will merge after those are fixed.

**Review by DonnieBLT — DISMISSED** [review]

File: blt/urls.py [hunk 14]
```diff
@@ -415,7 +415,8 @@
     re_path(r"^contributors/$", contributors_view, name="contributors"),
     path("company/", include("company.urls")),
     path("sponsor/", website.views.sponsor_view, name="sponsor"),
-    path("companies/", DomainListView.as_view(), name="domain_lists"),
+    path("companies/", DomainListView.as_view(), name="domain_list"),
```
Comment (DonnieBLT) [inline blt/urls.py:418]: Why did you change this?

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Please revert companies line

File: blt/urls.py (hunk 14 above)
Comment (Sarthak5598) [reply blt/urls.py]: Sorry! Changed it , Everything is working perfectly now

**Comment by DonnieBLT** [comment]
Looks like you have an old copy of the code. Can you please create your branch from main?

**Comment by DonnieBLT** [comment]
Closing all PRs created before the linting was done. Please recreate your PRs from the updated main branch.
